Assets/Scripts/Player/ShapeMatcher.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class ShapeMatcher : MonoBehaviour
+{
+    public PilarCheck[] puzzlePieces;
+    public bool puzzlePiecesComplete;
+    public bool allPilarsInPlace;
+    bool allPuzzlePiecesInPlace = false;
+    public PilarCheck pilarCheck;
```

Comment:
```suggestion
    [SerializeField] private Pillars[] pillars;
    [SerializeField] private UnityEvent onCompletion = new();
```
Simpele benamingen zijn beter leesbaar en alle booleans zijn nutteloos. we hoeven niet 4 booleans om deze class te markeren of iets klaar is. 
Ook UnityEvent gebruiken zodat we het samen met andere systemen kunnen gebruiken in een goede manier.